src/App.tsx

```diff
@@ -14,11 +15,87 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goodsToShow: string[],
+  hiddenStart: boolean,
+  reversed: boolean,
+  sortAlphabet: boolean,
+  sortLength: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goodsToShow: [],
+    hiddenStart: false,
+    reversed: false,
+    sortAlphabet: false,
+    sortLength: false,
```

Comment:
create a common property `sortBy`, because it can be much more sorting variants and it's not good practice to create a property for each